src/task/PrimeTask.php

```diff
@@ -230,25 +234,42 @@ private function updateDeadFiles($removed)
 
     /**
      * Constructs the SQL query and executes it.
-     * @param Node[] $new
+     * @param string[] $new
      */
-    private function insertNewFileFunctions(array $new)
+    private function insertNewFileFunctions(array $new): void
     {
         // early return when there is nothing to be added
         if (empty($new)) {
             return;
         }
-        $table  = $this->getFunctionsTable();
-        $db     = $this->getDb();
-        $values = [];
+        $table = $this->getFunctionsTable();
+        $db    = $this->getDb();
 
-        foreach ($new as $data) {
-            foreach ($data->getFileFunctions() as $file_function) {
-                $values[] = "(\"$file_function\", NOW())";
-            }
+        foreach ($new as $file_function) {
+            $query = $db->prepare("INSERT INTO $table (function, added_at) VALUES (:file_function, NOW())");
+            $query->bindParam(":file_function", $file_function);
+            $query->execute();
+        }
+    }
+
+
+    /**
+     * @param string[] $dead_functions
+     */
+    private function updateDeadFunctions(array $dead_functions): void
```

Comment:
`iterable $dead_functions`, because you only iterate over it, don't use it as hash map.